### DISCUSSION TITLE
essay/2024/12/23/writing-code

### DISCUSSION THREAD

**Discussion by giscus** [BenjaminKim/benjaminkim.github.io#249] [Comments]
# essay/2024/12/23/writing-code

남이 짠 코드를 읽는 건 재미없는 일입니다. 가끔 잘 짜인 코드를 볼 때면 기분이 좋았지만… 그런 일이 많지는 않았습니다. 잘 짠 코드라 하더라도 코드 타래를 끝도 없이 따라가며 의도를 파악하기까지의 순간은 고통 그 자체였습니다. 내가 코딩할 때나 재미를 느끼는 것입니다.

https://jeho.page/essay/2024/12/23/writing-code.html

**Comment by code-yeongyu** [comment]
늘 잘 읽고 있습니다.

최근 [커서 팀 인터뷰](https://youtu.be/7P-9i208LHw?si=m02ruvjwjO0Du33c&t=235) 를 보았는데요, 반복작업들이 줄어들고 코딩의 가장 재미있는 부분만 남은것 같다고 표현하더라고요. 저는 참 공감되었습니다. 내 머릿속에 명확히 있고 타이핑 치기 귀찮은것들은 온통 AI 한테 맡겨버리고, 진짜 어려운 고민들을 할 수 있는것이, 케이크에서 초콜릿 부분만 똑 떼어먹는 느낌이랄까요?!

최근 Cline 이라는 vscode extension 도 써보고 있습니다. 기본적으론 커서 컴포저 기능이랑 비슷한데요. 이 Cline 의 포크인 Roo Cline 이라는거엔 auto approve 기능이 있더라고요. 

예전엔 프로젝트 셋업하는게 너무 귀찮았는데 이제 그런건 명령 몇줄로 맡겨버리고 조금 수정해주면, 진짜 내가 좋아하는 장난감을 다룰 수 있어서 좋은것 같아요.

AI 와 싸워야하는 바둑과 달리 코딩은 팀전이니까 저는 더 든든하던데, 너무 똑똑해져서 제가 무임승차 팀원이 되기 전 까지는 최대한 친하게 지내보려고 합니다 ㅎㅎ

**Reply by BenjaminKim** [reply]
그렇게 바라볼 수도 있겠군요. 좀 더 열심히 써보면서 확인해봐야겠네요. 코딩의 재밌는 부분만 남아있다는 말이 사실이면 정말 기쁠 것 같습니다. 😁

**Comment by yuhaeun-la** [comment]
벤자민님의 글은 유독 기억에 오래 남고 자꾸 와서 읽고싶은 매력이 있습니다 ㅎㅎ 그 이유를 곰곰이 생각해보면, 글 속에 깃든 진솔함 때문 아닌가 생각이들어요 
아무튼 간에 저는 요즘 새로운 언어를 배울 때 커서한테 냅다 짜보라고 합니다. 그리고 이해가 안가게 짜놨으면 막 싸우기도 해요. 너 ~ 이 부분 왜이렇게 짰어 ? 비효율적이다. 하면서 제가 감히(?) 일침을 날릴 때도 있고, 커서한테 제 코드 평가해보라고도 합니다. ㅋㅋㅋㅋ 윗 분 말처럼 팀원이 생긴 느낌이어서 재밌어요. 종종 제가 지적하면 커서가 '아. 제가 그부분을 놓쳤네요' 할 때도 있는데, 회사에서 자꾸 실수하는 저 같아서 마음이 아프기도 한다는...😵‍💫😅

**Reply by BenjaminKim** [reply]
진솔함 때문에 기억에 오래 남다니 최고의 칭찬인걸요? 
저도 커서와 함께 사는 삶을 실천 중이에요. 처음 만난 순간부터 강렬하게 나는 이 친구보다 더 코딩을 잘 할수 없다는 걸 느꼈기 때문에 부하 부리듯이 하지 않고 부탁하는 말투로 사용 중입니다. ㅋㅋ
